packages/calcite-components/src/components/combobox/combobox.tsx

```diff
@@ -782,6 +816,107 @@ export class Combobox
     this.updateActiveItemIndex(targetIndex);
   }
 
+  private hideChip(chipEl: HTMLCalciteChipElement): void {
+    chipEl.classList.add(CSS.chipInvisible);
+  }
+
+  private showChip(chipEl: HTMLCalciteChipElement): void {
+    chipEl.classList.remove(CSS.chipInvisible);
+  }
+
+  private maxCompactBreakpoint: number;
+
+  @State() compactDisplayMode = false;
+
+  private refreshDisplayMode = async () => {
+    await componentLoaded(this);
+
+    if (isSingleLike(this.selectionMode)) {
+      return;
+    }
+
+    if (!this.textInput) {
+      return;
+    }
+
+    const {
+      allSelectedIndicatorChipEl,
+      chipContainerEl,
+      displayMode,
+      getSelectedItems,
+      hideChip,
+      placeholder,
+      selectedIndicatorChipEl,
+      showChip,
+      textInput,
+    } = this;
+
+    const chipContainerElGap = parseInt(getComputedStyle(chipContainerEl).gap.replace("px", ""));
+    const chipContainerElWidth = getElementWidth(chipContainerEl);
+    const { fontSize, fontFamily } = getComputedStyle(textInput);
+    const inputTextWidth = getTextWidth(placeholder, `${fontSize} ${fontFamily}`);
+    const inputWidth = (inputTextWidth || 48) + chipContainerElGap;
+    const allSelectedIndicatorChipElWidth = getElementWidth(allSelectedIndicatorChipEl);
+    const selectedIndicatorChipElWidth = getElementWidth(selectedIndicatorChipEl);
+    const largestSelectedIndicatorChipWidth = Math.max(
+      allSelectedIndicatorChipElWidth,
+      selectedIndicatorChipElWidth
+    );
+
+    // Determine if we are in Compact Display Mode
+    const newCompactBreakpoint = Math.round(
+      largestSelectedIndicatorChipWidth + chipContainerElGap + inputWidth
+    );
+    if (!this.maxCompactBreakpoint || this.maxCompactBreakpoint < newCompactBreakpoint) {
+      this.maxCompactBreakpoint = newCompactBreakpoint;
+    }
+    this.compactDisplayMode = chipContainerElWidth < this.maxCompactBreakpoint;
+
+    if (displayMode === "fit-to-line") {
+      const chipEls = Array.from(this.el.shadowRoot.querySelectorAll("calcite-chip")).filter(
+        (chipEl) => chipEl.hasAttribute("closable")
+      );
+
+      let availableHorizontalChipElSpace = Math.round(
+        chipContainerElWidth -
+          ((this.selectedHiddenChipsCount > 0 ? selectedIndicatorChipElWidth : 0) +
+            chipContainerElGap +
+            inputWidth +
+            chipContainerElGap)
+      );
+
+      // Refresh Selected Chips
+      chipEls.forEach((chipEl: HTMLCalciteChipElement) => {
+        if (chipEl.selected) {
+          const chipElWidth = getElementWidth(chipEl);
+          if (chipElWidth && chipElWidth < availableHorizontalChipElSpace) {
+            availableHorizontalChipElSpace -= chipElWidth + chipContainerElGap;
+            showChip(chipEl);
+          } else {
+            hideChip(chipEl);
+          }
+        } else {
```

Comment:
For readability, can you flip the order of these? This way the shorter block will be easier to spot and scan and the bigger one can follow. Bonus points for returning early.